### PR TITLE
disable upgrades to 26 for 32-bit

### PR DESCRIPTION
**Pull request by szaimen**
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by szaimen** [comment]
Documentation update is already there: https://github.com/nextcloud/documentation/pull/9274

**Review by artonge — APPROVED** [review]

**Comment by nickvergessen** [comment]
This does not block the updater (`updater/updater.phar` from release bundle).
I think we should have implemented it similarly to the PHP version check directly on the updater server (or should do that additionally)

**Comment by szaimen** [comment]
You mean something like this?
https://github.com/nextcloud/updater/pull/446

**Comment by szaimen** [comment]
Reviews are welcome :)

**Comment by nickvergessen** [comment]
No, more like https://github.com/nextcloud/updater_server/blob/master/config/config.php#L159-L190
So that in 32 bit you get back the last working version instead of a newer one together with a "please replace your hardware" message which comes with a lot of effort.

File: lib/private/Updater.php [hunk 1]
```diff
@@ -138,6 +139,9 @@ public function upgrade(): bool {
 
 		$success = true;
 		try {
+			if (PHP_INT_SIZE < 8 && Semver::satisfies($currentVersion, '> 25')) {
+				throw new HintException('You are running a 32-bit PHP version. Cannot upgrade to Nextcloud 26 and higher. Please switch to 64-bit PHP.');
```
Comment (nickvergessen) [inline lib/private/Updater.php:143]: Also this is not really "32-bit PHP"
In https://github.com/nextcloud/updater/pull/446/files#diff-7413d6453f901e939bbd840c8f0d1c7b20c2ca0e7f71741e4e07c6cf036f16c0R171 the wording is a bit better with "32-bit system"

**Comment by szaimen** [comment]
> No, more like https://github.com/nextcloud/updater_server/blob/master/config/config.php#L159-L190 (note the `minPHPVersion`) So that in 32 bit you get back the last working version instead of a newer one together with a "please replace your hardware" message which comes with a lot of effort.

Could you give me some more hints what would need to be done to implement this? I am honestly a bit scared of the updater_server code...

**Pull request merged**